metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py

```diff
@@ -130,74 +133,132 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
         ]
 
     def _get_datasets(self, catalog: Catalog) -> Iterable[Identifier]:
-        for namespace in catalog.list_namespaces():
-            yield from catalog.list_tables(namespace)
+        namespaces = catalog.list_namespaces()
+        LOGGER.debug(
+            f"Retrieved {len(namespaces)} namespaces, first 10: {namespaces[:10]}"
+        )
+        tables_count = 0
+        for namespace in namespaces:
+            tables = catalog.list_tables(namespace)
```

Comment:
Can this fail? If it can maybe we should handle it to not fail all the namespaces.

Reply:
Added handling of exceptions there.